scripts/tests/test_validate_docstrings.py

```diff
@@ -584,6 +587,29 @@ def prefix_pandas(self):
         pass
 
 
+class BadExamples(object):
+
+    def doctest(self):
```

Comment:
I think `doctest` is not a very descriptive name if this test is to check whether pep8 should fail because of unknown `np`. The rest of the example is very verbose to just check that. I'd use the description to explain what is this test about. And in the examples just with `>>> np.nan` could be enough.

Then, couple more tests could be added, to test the typical pep8 issues we have, like `>>> foo = 1+2` (missing spaces)...

Reply:
The reason to use flake8 to check for pep8 issues was to not have to write all these tests. flake8 should be tested enough to work properly. 

This test basically only verifies that flake8 is actually testing the docstrings.